### PR TITLE
Fix CI by ignoring failure of `stop-` jobs

### LINKED ISSUE #608: Investigate splitting up the publish_draft_releases workflow
[publish_draft_releases.yml](https://github.com/Manta-Network/Manta/blob/manta/.github/workflows/publish_draft_releases.yml) currently contains

- integration tests
- docker test
on normal PRs

and the draft release generation on release candidates.

I open the file, see 1152 lines and immediately feel regret for doing so.

To reduce complexity and in the spirit of single-responsibility we should refactor this into multiple files, so that `publish_draft_releases` does exactly what the name implies and nothing more

look into https://docs.github.com/en/actions/using-workflows/reusing-workflows for modularzation

### PR DESCRIPTION
**Pull request by Garandor**
stop-jobs run always even if their corresponding `start-` job has been skipped in order to avoid situations where we could get left-over zombie runners.

To not fail CI if a runner can't be stopped (e.g. because it has never been started), we ignore failure of stop- jobs

Signed-off-by: Adam Reif <Garandor@manta.network>

## Description

closes: #608

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [ ] Explicitly labelled `A-calamari`, `A-dolphin` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.
- If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. You can run the `metadata_diff.yml` workflow for help. If this number is updated, then the `spec_version` must also be updated
- Verify benchmarks & weights have been updated for any modified runtime logics

### CODE REVIEW

**Comment by Garandor** [comment]
Well, I was hoping the CI status would show success in this case, not the dreaded `X` for failure, guess this needs another solution, maybe we could add a second step to each stop job that just returns true and have `continue_on_error` on the aws stop step of the job.

@ghzlatarev could you maybe have a stab at this? The relevant doc is here: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error

**Review by Dengjianping — DISMISSED** [review]

**Comment by ghzlatarev** [comment]
> Well, I was hoping the CI status would show success in this case, not the dreaded `X` for failure, guess this needs another solution, maybe we could add a second step to each stop job that just returns true and have `continue_on_error` on the aws stop step of the job.
> 
> @ghzlatarev could you maybe have a stab at this? The relevant doc is here: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error

Will look into it some more, all of my ideas were not going to work as well :D

**Comment by ghzlatarev** [comment]
@Garandor we can go with always running the start/stop jobs and just skip the middle parts, like in my last commit.
That's going to resolve our problem without `continue-on-error: true` but it's a bit less efficient.
I'm ok with this, just copy the solution for the other jobs if you agree.

File: .github/workflows/integration_test_calamari.yml [hunk 1]
```diff
@@ -408,7 +408,7 @@ jobs:
           if [ $? == 1 ]; then echo "Failed to finalize the target block - 6"; exit 1; fi
   # HELPER JOBS BELOW
   start-node-builder-current:
-    if: contains(github.event.pull_request.labels.*.name, 'A-calamari' || github.ref == 'refs/heads/manta')
+    if: ${{ always() }}
```
Comment (ghzlatarev) [inline .github/workflows/integration_test_calamari.yml]: revert this change

File: .github/workflows/integration_test_calamari.yml [hunk 2]
```diff
@@ -444,8 +445,10 @@ jobs:
           aws-region: ${{ needs.start-node-builder-current.outputs.aws-region }}
           runner-label: ${{ needs.start-node-builder-current.outputs.runner-label }}
           aws-instance-id: ${{ needs.start-node-builder-current.outputs.aws-instance-id }}
+      - name: discard stopper success/failure
+        run: true
   start-calamari-integration-tester:
-    if: contains(github.event.pull_request.labels.*.name, 'A-calamari' || github.ref == 'refs/heads/manta')
+    if: ${{ always() }}
```
Comment (ghzlatarev) [inline .github/workflows/integration_test_calamari.yml]: and this

File: .github/workflows/integration_test_calamari.yml [hunk 3]
```diff
@@ -481,8 +485,10 @@ jobs:
           aws-region: ${{ needs.start-calamari-integration-tester.outputs.aws-region }}
           runner-label: ${{ needs.start-calamari-integration-tester.outputs.runner-label }}
           aws-instance-id: ${{ needs.start-calamari-integration-tester.outputs.aws-instance-id }}
+      - name: discard stopper success/failure
+        run: true
   start-docker-image-tester:
-    if: contains(github.event.pull_request.labels.*.name, 'A-calamari' || github.ref == 'refs/heads/manta')
+    if: ${{ always() }}
```
Comment (ghzlatarev) [inline .github/workflows/integration_test_calamari.yml]: and this

**Review by ghzlatarev — APPROVED** [review]

**Comment by ghzlatarev** [comment]
nice

**Review by flame4 — APPROVED** [review]

**Review by Dengjianping — APPROVED** [review]
